src/librustc/ty/error.rs

```diff
@@ -193,10 +192,10 @@ impl<'a, 'gcx, 'lcx, 'tcx> ty::TyS<'tcx> {
             ty::TyAdt(def, _) => format!("{} `{}`", def.descr(), tcx.item_path_str(def.did)),
             ty::TyForeign(def_id) => format!("extern type `{}`", tcx.item_path_str(def_id)),
             ty::TyArray(_, n) => {
-                if let ConstVal::Integral(ConstInt::Usize(n)) = n.val {
-                    format!("array of {} elements", n)
-                } else {
-                    "array".to_string()
+                match n.val {
```

Comment:
I think it could be entirely unevaluable. If it is evaluable, it'll produce the `Bytes` Primval, if not, we can't print anything anyway. Trying to print weird things like this being a pointer to a string literal also makes little sense imo.

Reply:
I just mean that since this is an array length, it could use a `-> Option<u64>` method or something like that, to avoid matching itself on `ConstVal`.

Reply:
done